### PR TITLE
fix(ci): add `ci-dev` profile flags to `main-cron` config

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

- Use `ci-dev` flags for our cron workflow as well, otherwise `streaming e2e` will timeout (exits with 124).
- Configure `ci-dev` profile for `e2e-source-test` in cron config as well.

See failing cron report here: https://buildkite.com/singularity-data/main-cron/builds/93#01829487-b997-47a5-be2f-71e98fa13f32

## Checklist

~~- [x] I have written necessary rustdoc comments~~
~~- [x] I have added necessary unit tests and integration tests~~
~~- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)~~

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: ci/workflows/main-cron.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ auto-retry: &auto-retry
 
 steps:
   - label: "build"
-    command: "ci/scripts/build.sh -t debug -p dev"
+    command: "ci/scripts/build.sh -t debug -p ci-dev"
```
Comment (BugenZhao) [inline ci/workflows/main-cron.yml:24]:
```suggestion
    command: "ci/scripts/build.sh -t ci-dev -p ci-dev"
```

**Review by BugenZhao — COMMENTED** [review]
LGTM. BTW, why are we not using `release` profile for main-cron? 🤔 cc @huangjw806 @skyzh

**Review by BugenZhao — APPROVED** [review]

**Comment by skyzh** [comment]
> LGTM. BTW, why are we not using `release` profile for main-cron? 🤔 cc @huangjw806 @skyzh

I think we should, let's fix this next time 🤣

**Comment by BugenZhao** [comment]
> > LGTM. BTW, why are we not using `release` profile for main-cron? 🤔 cc @huangjw806 @skyzh
> 
> I think we should, let's fix this next time 🤣

Using `ci-dev` in main-cron also LGTM, since we're already using `release` for main. Adding test analytics for main-cron with `ci-dev` can also make us easier to check performance degradation.

**Pull request merged**